spec/Section 3 -- Type System.md

```diff
@@ -1340,14 +1357,281 @@ Union type extensions have the potential to be invalid if incorrectly defined.
+
+The result of coercion is an unordered map with an entry for exactly one
+member both defined by the tagged type and present in the input. The resulting
+map is constructed with the following rules:
+
+* If more than one field is defined in the input, an error must be thrown.
+
+* For the single field defined in the input, if the value is {null} and the
+  field's type is a non-null type, an error must be thrown.
+
+* If a literal value is provided for an input object field, an entry in the
+  coerced unordered map is given the result of coercing that value according
+  to the input coercion rules for the type of the tagged member for that field.
+
+* If a variable is provided for an input object field, the runtime value of that
+  variable must be used. If the runtime value is {null} and the tagged member
+  type is non-null, a field error must be thrown. If no runtime value is
+  provided, the variable definition's default value should be used. If the
+  variable definition does not provide a default value, the field should be
+  treated as if it was not specified.
+
+* No entry should be added to the unordered map for any other members defined
+  on the Tagged type.
+
+Following are examples of input coercion for a Tagged type with a
+`String` member `a` and a required (non-null) `Int!` member `b`:
+
+```graphql example
+tagged ExampleInputTagged {
+  a: String
+  b: Int!
+}
+```
+
+Literal Value            | Variables               | Coerced Value
+------------------------ | ----------------------- | ---------------------------
+`{ a: "abc", b: 123 }`   | `{}`                    | Error: Exactly one key must be specified
+`{ a: null, b: 123 }`    | `{}`                    | Error: Exactly one key must be specified
+`{ b: 123 }`             | `{}`                    | `{ b: 123 }`
+`{ a: $var, b: 123 }`    | `{ var: null }`         | Error: Exactly one key must be specified
+`{ a: $var, b: 123 }`    | `{}`                    | `{ b: 123 }`
+`{ b: $var }`            | `{ var: 123 }`          | `{ b: 123 }`
+`$var`                   | `{ var: { b: 123 } }`   | `{ b: 123 }`
+`"abc123"`               | `{}`                    | Error: Incorrect value
+`$var`                   | `{ var: "abc123" } }`   | Error: Incorrect value
+`{ a: "abc", b: "123" }` | `{}`                    | Error: Exactly one key must be specified
+`{ b: "123" }`           | `{}`                    | Error: Incorrect value for member {b}
+`{ a: "abc" }`           | `{}`                    | `{ a: "abc" }`
+`{ b: $var }`            | `{}`                    | Error: No keys were specified
+`$var`                   | `{ var: { a: "abc" } }` | `{ a: "abc" }`
+`{ a: "abc", b: null }`  | `{}`                    | Error: Exactly one key must be specified
+`{ b: $var }`            | `{ var: null }`         | Error: {b} must be non-null.
+`{ b: 123, c: "xyz" }`   | `{}`                    | Error: Unexpected member {c}
```

Comment:
this example implies that an implementation would check that each member field is valid before checking that only one field was provided, I'm not confident that's what we should suggest

Reply:
Good catch; that was unintended :+1:

Reply:
I'm changing this to `Error: Exactly one key must be specified`